src/test_l1_tvSettings.c

```diff
@@ -13824,120 +14105,120 @@ void test_l1_tvSettings_negative_SaveCMS (void)
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 
 	/* Step 16: Calling tvsettings SaveCMS and expecting the API to return tvERROR_INVALID_PARAM */
-	result = SaveCMS((tvVideoSrcType_t)Configfile.videoSrcStruct.videoSourceValue[0],Configfile.picmodeStruct.pqValue[0],(tvVideoFormatType_t)Configfile.videoFormtStruct.videoFormatValue[0],COMP_HUE, (tvDataComponentColor_t)Configfile.componentColor.modeId[0],101);
+	result = SaveCMS((tvVideoSrcType_t)videoSource,pqValue,(tvVideoFormatType_t)videoFormat,COMP_HUE, (tvDataComponentColor_t)Configfile.componentColor.modeId[0],101);
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 
 	/* Step 17: Calling tvsettings SaveCMS and expecting the API to return tvERROR_INVALID_PARAM */
-	result = SaveCMS((tvVideoSrcType_t)Configfile.videoSrcStruct.videoSourceValue[0],Configfile.picmodeStruct.pqValue[0],(tvVideoFormatType_t)Configfile.videoFormtStruct.videoFormatValue[0],COMP_LUMA, (tvDataComponentColor_t)Configfile.componentColor.modeId[0],-1);
+	result = SaveCMS((tvVideoSrcType_t)videoSource,pqValue,(tvVideoFormatType_t)videoFormat,COMP_LUMA, (tvDataComponentColor_t)Configfile.componentColor.modeId[0],-1);
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 
 	/* Step 18: Calling tvsettings SaveCMS and expecting the API to return tvERROR_INVALID_PARAM */
-	result = SaveCMS((tvVideoSrcType_t)Configfile.videoSrcStruct.videoSourceValue[0],Configfile.picmodeStruct.pqValue[0],(tvVideoFormatType_t)Configfile.videoFormtStruct.videoFormatValue[0],COMP_LUMA, (tvDataComponentColor_t)Configfile.componentColor.modeId[0],31);
+	result = SaveCMS((tvVideoSrcType_t)videoSource,pqValue,(tvVideoFormatType_t)videoFormat,COMP_LUMA, (tvDataComponentColor_t)Configfile.componentColor.modeId[0],31);
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 
 	/* Step 19: Calling tvsettings SaveCMS and expecting the API to return tvERROR_INVALID_PARAM */
 	for( i =VIDEO_SOURCE_ALL ; i < VIDEO_SOURCE_MAX; i++)
 	{
 		SupportAvailable = false;
-		for( j =0 ; j < Configfile.videoSrcStruct.size; j++)
+		for( j =0 ; j < videoSrcCount; j++)
 		{
-			if(Configfile.videoSrcStruct.videoSourceValue[j] == i)
+			if(videoSource == i)
 			{
 				SupportAvailable = true;
 				break;
 			}
 		}
 
 		if(!SupportAvailable){
-			result = SaveCMS((tvVideoSrcType_t)i,Configfile.picmodeStruct.pqValue[0],(tvVideoFormatType_t)Configfile.videoFormtStruct.videoFormatValue[0],(tvComponentType_t)Configfile.componentColor.modeId[0], (tvDataComponentColor_t)Configfile.componentColor.modeId[0],1);
+			result = SaveCMS((tvVideoSrcType_t)i,pqValue,(tvVideoFormatType_t)videoFormat,(tvComponentType_t)Configfile.componentColor.modeId[0], (tvDataComponentColor_t)Configfile.componentColor.modeId[0],1);
 			UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
```

Comment:
Configfile.componentColor.modeId is still present

Reply:
Replaced